variables.tf

```diff
@@ -187,6 +188,7 @@ variable "node_pools" {
     preemptible                 = false
     max_surge                   = 1
     max_unavailable             = 0
+    enable_shielded_nodes       = null
```

Comment:
Why is it provided null if it's actually of bool type?

Reply:
because of how the DEFAULTS of  `shielded_instance_config` are configured in terraform.

The 2 attributes under `shielded_instance_config` sets a MIX of `true` & `false` by default by GCP.

If we set our variable default to either `true` or `false`, it will end up impacting at least one of the 2 attributes which are both `true` & `false`. Hence, by default, we keep it null, so that we can let `coalesce()` determine their preferred defaults if `null` is specified - which is what will ensure backward-compatibility.

Hence added the comment 'set default values as per the defaults stated in google provider'. in that portion.